File: benchmark/java/src/main/java/com/benchmark/services/LdapService.java
package com.benchmark.services;

import javax.naming.directory.DirContext;
import javax.naming.directory.InitialDirContext;
import javax.naming.directory.SearchControls;
import javax.naming.directory.NamingEnumeration;

public class LdapService {
    public void searchUser(String username) {
        try {
            DirContext ctx = new InitialDirContext();
            SearchControls controls = new SearchControls();
            
            // LDAP Injection [CWE-90]
            String filter = "(uid=" + username + ")";
            NamingEnumeration results = ctx.search("ou=users,dc=example,dc=com", filter, controls);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: benchmark/csharp/Services/LdapService.cs
using System.DirectoryServices;

namespace Benchmark.CSharp.Services
{
    public class LdapService
    {
        public void FindUser(string username)
        {
            // LDAP Injection [CWE-90]
            DirectoryEntry entry = new DirectoryEntry("LDAP://OU=Users,DC=example,DC=com");
            DirectorySearcher searcher = new DirectorySearcher(entry);
            
            searcher.Filter = "(sAMAccountName=" + username + ")";
            SearchResult result = searcher.FindOne();
        }
    }
}
